packages/connect-web/src/popup/index.ts

```diff
@@ -281,12 +281,15 @@ export class PopupManager extends EventEmitter {
             // send ConnectSettings to popup
             // note this settings and iframe.ConnectSettings could be different (especially: origin, popup, webusb, debug)
             // now popup is able to load assets
-        } else if (data.type === POPUP.CANCEL_POPUP_REQUEST || data.type === UI.CLOSE_UI_WINDOW) {
-            this.close();
+        } else if (
+            data.type === POPUP.CANCEL_POPUP_REQUEST
+            //  || data.type === UI.CLOSE_UI_WINDOW // Ignoring because it is always sent with CANCEL_POPUP_REQUEST
+        ) {
+            this.close(message);
         }
     }
 
-    close() {
+    close(message?: MessageEvent) {
```

Comment:
this method is called couple of times without argument in `packages/connect-web/src/popup/index.ts` and `packages/connect-web/src/index.ts`. Don't we want to keep popup open also in those cases?

Reply:
Not applicable anymore since using RESPONSE_EVENT is not required change.